controller/commentsController.js

```diff
@@ -271,3 +272,29 @@ exports.deleteComment = async (req, res, next) => {
     return next(error);
   }
 };
+exports.getSingleComment = async (req, res, next) => {
+  const { refId } = req.query;
+  const commentId = req.params.commentId;
+  const applicationId = req.headers.token; //this will be retrieved from decoded api token after full auth implementation
+  const query = { applicationId: applicationId, _id: commentId };
+  if (refId) query.refId = refId;
+  try {
+    await Comments.find(query)
+      .populate("replies")
+      .then((comment) => {
+        res.status(200).json({
+          status: "success",
+          message: `Comment Retrieved Successfully`,
+          query: query,
+          data: comment,
+        });
+      })
+      .catch(next);
```

Comment:
Please handle the error properly in the catch block before passing to default errorhandler so as to have an intuitive error message instead of a mongoose generated error message. I'd suggest creating a new `CustomError` object and passing that to next instead